pkg/update/update_cockroach_version.go

```diff
@@ -152,32 +144,32 @@ func updateClusterStatefulSets(
 // waitUntilAllPodsReady waits until all pods in all statefulsets are in the
 // ready state. The ready state implies all nodes are passing node liveness.
 func makeWaitUntilAllPodsReadyFunc(
+	ctx context.Context,
 	clientset kubernetes.Interface,
 	podUpdateTimeout time.Duration,
 	maxPodPollingInterval time.Duration,
 	stsName string,
 	ns string,
-) func(ctx context.Context, l *zap.Logger) error {
-	return func(ctx context.Context, l *zap.Logger) error {
+) func(ctx context.Context, l logr.Logger) error {
+	return func(ctx context.Context, l logr.Logger) error {
 
 		l.Info("waiting until all pods are in the ready state")
 		f := func() error {
 
 			sts, err := clientset.AppsV1().StatefulSets(ns).Get(ctx, stsName, metav1.GetOptions{})
 			if err != nil {
-				l.Warn("could not find crdb sts")
-				return errors.Wrapf(err, "could not find crdb sts for %s", stsName)
+				return handleStsError(err, l, stsName, ns)
 			}
 			got := int(sts.Status.ReadyReplicas)
 			// TODO need to test this
 			// we could also use the number of pods defined by the operator
 			numCRDBPods := int(sts.Status.Replicas)
 			if got != numCRDBPods {
-				l.Sugar().Warnf("number of ready replicas is %v, not equal to num CRDB pods %v", got, numCRDBPods)
-				return fmt.Errorf("number of ready replicas is %v, not equal to num CRDB pods %v", got, numCRDBPods)
+				l.Error(err, fmt.Sprintf("number of ready replicas is %v, not equal to num CRDB pods %v", got, numCRDBPods))
+				return err
 			}
 
-			l.Info("all replicas are ready")
+			l.Info("all replicas are ready makeWaitUntilAllPodsReadyFunc update_cockroach_version.go")
```

Comment:
Is there not an option to include the lineno and file log messages come from?

Reply:
I have no idea